src/library/scanner/libraryscanner.cpp

```diff
@@ -573,6 +574,17 @@ void LibraryScanner::slotTrackExists(const QString& trackPath) {
 
 void LibraryScanner::slotAddNewTrack(const QString& trackPath) {
     // kLogger.debug() << "slotAddNewTrack" << trackPath;
+    if (!m_scannerGlobal || m_scannerGlobal->shouldCancel()) {
+        // Fix/workaround for Cancel not cancelling the entire scan process
+        // https://github.com/mixxxdj/mixxx/issues/14940
+        // Pretty quickly after starting the scan, many ImportFilesTask queue
+        // many addNewTrack() signals connected to this slot. When cancelling the
+        // scan via Cancel button in the progress dialog, all signals are usually
+        // already queued, hence Cancel has no effect on these calls and Mixxx
+        // keeps adding/analyzing tracks as if nothing happened.
+        // Simply abort here does the trick.
+        return;
+    }
```

Comment:
Note: we might also try `disconnect()`, but I didn't manage to disconnect just `slotAddNewTrack()`